### PR TITLE
TEP-0142: Add syntax for providing params to StepActions

### PR DESCRIPTION
**Pull request by chitrangpatel**
Following the previous [PR](https://github.com/tektoncd/pipeline/pull/7317), which introduced Params to the `StepAction` CRD, this PR integrates `param` usage between `Steps` and `StepActions`. This PR adds the necessary syntax to `Steps` to pass `params` to `StepActions`. This work is part of issue https://github.com/tektoncd/pipeline/issues/7259. The actual resolution of the params and passing them to `StepActions` will be completed in PR https://github.com/tektoncd/pipeline/pull/7332.

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```
/kind feature

### CODE REVIEW

File: pkg/apis/pipeline/v1/task_validation.go [hunk 1]
```diff
@@ -300,6 +300,12 @@ func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.Fi
 			})
 		}
 	} else {
+		if len(s.Params) > 0 {
+			errs = errs.Also(&apis.FieldError{
+				Message: "params cannot be used without Ref",
+				Paths:   []string{"params"},
+			})
+		}
```
Comment (jerop) [inline pkg/apis/pipeline/v1/task_validation.go:308]: please update the docs with this requirement

Comment (jerop) [inline pkg/apis/pipeline/v1/task_validation.go:308]: please update the docs with this requirement

File: pkg/apis/pipeline/v1beta1/container_types.go [hunk 2]
```diff
@@ -232,6 +232,10 @@ type Step struct {
 	// Contains the reference to an existing StepAction.
 	//+optional
 	Ref *Ref `json:"ref,omitempty"`
+	// Parameters declares parameters passed to this step action.
```
Comment (jerop) [inline pkg/apis/pipeline/v1beta1/container_types.go:235]:
```suggestion
	// Params declares Parameters passed to this StepAction.
```

File: pkg/apis/pipeline/v1/task_validation.go (hunk 1 above)
Comment (chitrangpatel) [reply pkg/apis/pipeline/v1/task_validation.go:308]: done. PTAL `docs/stepactions.md`

File: pkg/apis/pipeline/v1beta1/container_types.go (hunk 2 above)
Comment (chitrangpatel) [reply pkg/apis/pipeline/v1beta1/container_types.go]: fixed!

**Review by jerop — APPROVED** [review]
🎉

**Comment by jerop** [comment]
@tektoncd/core-maintainers @tektoncd/core-collaborators please take a look

**Review by QuanZhang-William — APPROVED** [review]

**Review by Yongxuanzhang — APPROVED** [review]
lgtm!

**Pull request merged**